cocos/platform/CCImage.cpp

```diff
@@ -2165,10 +2170,10 @@ bool Image::initWithRawData(const unsigned char * data, ssize_t dataLen, int wid
         _height   = height;
         _width    = width;
         _hasPremultipliedAlpha = preMulti;
-        _renderFormat = Texture2D::PixelFormat::RGBA8888;
+        _renderFormat = renderFormat;
 
-        // only RGBA8888 supported
-        int bytesPerComponent = 4;
+        // only RGBA8888, BGRA8888 and RGB888 supported
+        int bytesPerComponent = (_renderFormat == Texture2D::PixelFormat::RGB888) ? 3 : 4;
```

Comment:
This function has *never* had any protection from people passing in the wrong data.  And I am not attempting to add any.  Only to support some of the more common "raw" formats.